ios/browser/api/history/brave_history_api.mm

```diff
@@ -109,6 +109,127 @@ - (NSDate*)dateAdded {
+                       hostOnly:NO
+              duplicateHandling:HistoryDuplicateHandlingIOSRemoveAll
+                      beginDate:nil
+                        endDate:nil];
+}
+
+- (instancetype)initWithMaxCount:(NSUInteger)maxCount
+               duplicateHandling:
+                   (HistoryDuplicateHandlingIOS)duplicateHandling {
+  return [self initWithMaxCount:maxCount
+                       hostOnly:NO
+              duplicateHandling:duplicateHandling
+                      beginDate:nil
+                        endDate:nil];
+}
+
+- (instancetype)initWithMaxCount:(NSUInteger)maxCount
+                        hostOnly:(BOOL)hostOnly
+               duplicateHandling:(HistoryDuplicateHandlingIOS)duplicateHandling
+                       beginDate:(nullable NSDate*)beginDate
+                         endDate:(nullable NSDate*)endDate {
+  if ((self = [super init])) {
+    [self setMaxCount:maxCount];
+    [self setHostOnly:hostOnly];
+    [self setDuplicateHandling:duplicateHandling];
+    if (beginDate) {
+      [self setBeginDate:beginDate];
+    }
+    if (endDate) {
+      [self setEndDate:endDate];
+    }
+  }
+  return self;
+}
+
+- (void)setMaxCount:(NSUInteger)maxCount {
+  max_count_ = static_cast<int>(maxCount);
+}
+
+- (NSUInteger)maxCount {
+  return static_cast<NSUInteger>(max_count_);
+}
+
+- (void)setHostOnly:(BOOL)hostOnly {
+  host_only_ = hostOnly;
+}
+
+- (BOOL)hostOnly {
+  return host_only_;
+}
+
+- (void)setBeginDate:(NSDate*)beginDate {
+  begin_time_ = base::Time::FromNSDate(beginDate);
+}
+
+- (NSDate*)beginDate {
+  return begin_time_.ToNSDate();
+}
+
+- (void)setEndDate:(NSDate*)endDate {
+  end_time_ = base::Time::FromNSDate(endDate);
+}
+
+- (NSDate*)endDate {
+  return end_time_.ToNSDate();
+}
+
+- (void)setDuplicateHandling:(HistoryDuplicateHandlingIOS)duplicateHandling {
+  switch (duplicateHandling) {
+    case HistoryDuplicateHandlingIOSRemoveAll:
+      duplicate_policy_ =
+          history::QueryOptions::DuplicateHandling::REMOVE_ALL_DUPLICATES;
+      break;
+    case HistoryDuplicateHandlingIOSRemovePerDay:
+      duplicate_policy_ =
+          history::QueryOptions::DuplicateHandling::REMOVE_DUPLICATES_PER_DAY;
+      break;
+    case HistoryDuplicateHandlingIOSKeepAll:
+      duplicate_policy_ =
+          history::QueryOptions::DuplicateHandling::KEEP_ALL_DUPLICATES;
+      break;
+  }
+}
+
+- (HistoryDuplicateHandlingIOS)duplicateHandling {
+  switch (duplicate_policy_) {
+    case history::QueryOptions::DuplicateHandling::REMOVE_ALL_DUPLICATES:
+      return HistoryDuplicateHandlingIOSRemoveAll;
+    case history::QueryOptions::DuplicateHandling::REMOVE_DUPLICATES_PER_DAY:
+      return HistoryDuplicateHandlingIOSRemovePerDay;
+    case history::QueryOptions::DuplicateHandling::KEEP_ALL_DUPLICATES:
+      return HistoryDuplicateHandlingIOSKeepAll;
+  }
+}
+
+- (history::QueryOptions)queryOptions {
+  history::QueryOptions options;
+  options.max_count = max_count_;
```

Comment:
These options should all be set directly from the Objective-C++ properties instead of having duplicate variables per property.